### PR TITLE
BAH-3127 | Updated Readme and Implemented Basic Github Action Build Workflow

### PR DESCRIPTION
**Pull request by Aniket28dot**
### Readme and Build Workflow Updates:
- Java Version, Maven Version
- WhatsApp API Set Up steps
- Webhook Configuration Steps
- Basic GitHub action workflow that performs build of the Microservice.

Jira Ticket: [link](https://bahmni.atlassian.net/browse/BAH-3127)

### CODE REVIEW

File: .DS_Store
Comment (rahu1ramesh) [inline .DS_Store:1]: Please add `.DS_Store` to the `.gitignore`. If you don't have a `.gitignore`, create one and add the files and folders that need not be committed.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Build package
+on:
+  push:
+    branches: [ main ]
+  workflow_dispatch:
```
Comment (rahu1ramesh) [inline .github/workflows/build.yml]: You can remove the `workflow_dispatch` for now. You can read further about `workflow_dispatch` [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch).

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  build-publish-package:
+    name: Build package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 20.0.1
```
Comment (rahu1ramesh) [inline .github/workflows/build.yml]: The choice of `JDK 20.0.1` for this Pull Request needs clarification. If the project's Java requirements do not specifically necessitate `JDK 20.0.1`, it is recommended to use `JDK 1.8` instead.
Please provide an explanation for the JDK selection or consider using `JDK 1.8`.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: 20.0.1
```
Comment (rahu1ramesh) [inline .github/workflows/build.yml]: The choice of `JDK 20.0.1` for this Pull Request needs clarification. If the project's Java requirements do not specifically necessitate `JDK 20.0.1`, it is recommended to use `JDK 1.8` instead.
Please provide an explanation for the JDK selection or consider using `JDK 1.8`.

File: .mvn/wrapper/maven-wrapper.jar
Comment (rahu1ramesh) [inline .mvn/wrapper/maven-wrapper.jar:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: .mvn/wrapper/maven-wrapper.properties
Comment (rahu1ramesh) [inline .mvn/wrapper/maven-wrapper.properties:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,29 @@
 # whatsapp-bahmni-service
 A microservice for Whatsapp to communicate to Bahmni (for Appointment booking) by Patients.
+
+# Java Version
+java version 20.0.1
```
Comment (rahu1ramesh) [inline README.md]: The choice of `JDK 20.0.1` for this Pull Request needs clarification. If the project's Java requirements do not specifically necessitate `JDK 20.0.1`, it is recommended to use `JDK 1.8` instead.
Please provide an explanation for the JDK selection or consider using `JDK 1.8`.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,29 @@
...
+
+# Maven Version
```
Comment (rahu1ramesh) [inline README.md]: Kindly refrain from adding separate sub headings for java and maven version requirements. Add a sub heading called `Requirements` and including the necessary details under it.

File: README.md
Comment (rahu1ramesh) [inline README.md:1]: Please do add a section on how to clone the repo and set up the project. You can refer this [README](https://github.com/Bahmni/bahmni-core/tree/master#clone-the-repository-and-build-the-omod) for the same.

File: mvnw
Comment (rahu1ramesh) [inline mvnw:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: mvnw.cmd
Comment (rahu1ramesh) [inline mvnw.cmd:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: pom.xml
Comment (rahu1ramesh) [inline pom.xml:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: src/main/java/org/bahmni/whatsapp/appointments/WhatsAppAPIController.java
Comment (rahu1ramesh) [inline src/main/java/org/bahmni/whatsapp/appointments/WhatsAppAPIController.java:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: src/main/java/org/bahmni/whatsapp/appointments/WhatsappBahmniServiceApplication.java
Comment (rahu1ramesh) [inline src/main/java/org/bahmni/whatsapp/appointments/WhatsappBahmniServiceApplication.java:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: src/main/resources/application.properties
Comment (rahu1ramesh) [inline src/main/resources/application.properties:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: src/test/java/org/bahmni/whatsapp/appointments/WhatsAppAPIControllerTest.java
Comment (rahu1ramesh) [inline src/test/java/org/bahmni/whatsapp/appointments/WhatsAppAPIControllerTest.java:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

File: src/test/java/org/bahmni/whatsapp/appointments/WhatsappBahmniServiceApplicationTests.java
Comment (rahu1ramesh) [inline src/test/java/org/bahmni/whatsapp/appointments/WhatsappBahmniServiceApplicationTests.java:1]: These changes are not part of the card BAH-3127, add a card for what this changes are for and then kindly raise another PR for the same.

**Review by rahu1ramesh — CHANGES_REQUESTED** [review]

File: .DS_Store
Comment (Aniket28dot) [reply .DS_Store:1]: sure will add .DS_Store in git ignore.

File: README.md (hunk 5 above)
Comment (Aniket28dot) [reply README.md]: yes, added the Requirements sections now.

File: README.md
Comment (Aniket28dot) [reply README.md:1]: yes, added this section for cloning the repo.

**Review by rahu1ramesh — APPROVED** [review]

**Pull request merged**